### DISCUSSION TITLE
Tesla Powerwall and Curtailment - works wiith pyPowewall?

### DISCUSSION THREAD

**Discussion by justinwunda** [amberelectric/public-api#234] [General]
Hi team, firstly, apologies if this is the wrong forum - it's possible this question should be directed to general Amber support, but as it involves some python tools I thought maybe here would be better.

It is understood that Amber does not currently support curtailment (throttling exports when FiT is negative) on a tesla system - at least it doesn't with my Powerwall 3. Yet I've been playing around with pyPowerwall (a python tool for managing your PW, available elsewhere on github), and through Cloud access (i.e. pyPowerwall interacts with my PW3 via my tesla account) I am able to set the "gridexport" mode to "never", thus completely halting grid exports.

My quetion is why is it not possible for Amber to do this? I have written a simple script that detects negative FiT's and sets gridexport to never, then when FiT's go positive it returns to normal operation. It seems to work fine. 

All I can think is that maybe Tesla does not allow retailers to alter this setting? Which seems strange because tesla allows other access to retailers that are not available to me as the end user, such as charging and discharging to and from the grid.

Thanks for any input.... apologies again if this is creating clutter as it's not about the Amber API

**Comment by madpilot** [comment]
We haven't received updated docs for doing it yet (our API is different to the API you are using).

We'll follow up again in the new year.

**Reply by justinwunda** [reply]
Fair enough - thanks for the fast reply! You guys are doing great work BTW, really loving Amber so far!
